### PR TITLE
Added blob NFS and some other validation fixes.

### PR DESCRIPTION
**Pull request by brpanask**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 1]
```diff
@@ -642,16 +636,13 @@
...
           },
           "202": {
-            "description": "Cache has started the transition to Active. Poll the Cache's state field to monitor.",
-            "schema": {}
+            "description": "Cache has started the transition to Active. Poll the Cache's state field to monitor."
           },
           "204": {
-            "description": "Restarted VMs associated with the cache.",
-            "schema": {}
+            "description": "Restarted VMs associated with the cache."
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:651]: Should this match 200's description? Not sure there is a must-have distinction.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 2]
```diff
@@ -706,16 +697,13 @@
...
           },
           "202": {
-            "description": "Cache has started the transition to Stopped. Poll the Cache's state field to monitor.",
-            "schema": {}
+            "description": "Cache has started the transition to Stopped. Poll the Cache's state field to monitor."
           },
           "204": {
-            "description": "Stopped VMs associated with the cache.",
-            "schema": {}
+            "description": "Stopped VMs associated with the cache."
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:712]: Similar comment re: matching 200's description.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 3]
```diff
@@ -1037,16 +1022,13 @@
         ],
         "responses": {
           "201": {
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: 201 is Created, which doesn't make sense for an upgrade action. Should this be 200?

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 4]
```diff
@@ -1545,57 +1529,60 @@
...
+            "network",
+            "host"
+          ],
+          "x-ms-enum": {
+            "name": "NfsAccessRuleScope",
+            "modelAsString": true
+          }
+        },
+        "filter": {
+          "description": "Filter applied to this rule: format depends on scope.  Default: unused, network: cidr (eg 10.99.1.0/24), host: ip address or FQDN",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Nitpicks: network: CIDR (e.g., 10.99.1.0/24), host: IP address or FQDN

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 5]
```diff
@@ -1545,57 +1529,60 @@
...
+            "ro",
+            "rw"
+          ],
+          "x-ms-enum": {
+            "name": "NfsAccessRuleAccess",
+            "modelAsString": true
+          }
+        },
+        "suid": {
+          "description": "Allow suid semantics.",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Nitpick: SUID

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 6]
```diff
@@ -1839,6 +1831,19 @@
         }
       }
     },
+    "BlobNfsTarget": {
+      "description": "Properties pertained to BlobNfsTarget",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Nitpick: pertaining

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 7]
```diff
@@ -1545,57 +1529,60 @@
...
+          "description": "Allow mounts below the junction.",
+          "type": "boolean"
+        },
+        "rootSquash": {
+          "description": "Map root accesses to anonymousUID and anonymousGID.",
+          "type": "boolean"
+        },
+        "anonymousUID": {
+          "description": "UID value that replaces 0 when rootSquash is true.",
+          "type": "string",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1593]: Just checking on this type. Does the RP expect a string or an integer? I believe UID/GID are always integers.

CC: @nydickmsft

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 8]
```diff
@@ -1839,6 +1831,19 @@
...
     },
+    "BlobNfsTarget": {
+      "description": "Properties pertained to BlobNfsTarget",
+      "properties": {
+        "target": {
+          "description": "Resource ID of storage container.",
+          "$ref": "#/definitions/URLString"
+        },
+        "usageModel": {
+          "description": "Identifies the primary usage model to be used for this Storage Target. Get choices from .../usageModels",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Nitpick: "primary" implies there's a secondary (or non-primary). I think "primary" can be removed.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json [hunk 9]
```diff
@@ -1839,6 +1831,19 @@
...
+          "type": "string"
+        }
+      }
+    },
     "ResourceName": {
       "description": "Schema for the name of resources served by this provider. Note that objects will contain an odata @id annotation as appropriate. This will contain the complete URL of the object. These names are case-preserving, but not case sensitive.",
       "pattern": "^[-0-9a-zA-Z_]{1,80}$",
```
Comment (omzevall) [inline specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1864]: I know it's not part of this PR, but this regex is overly permissive. Allowed names:

- single character: -, _, or any alphanum char
- string of hyphens and/or underscores, no alphanum chars

There's an internal bug regarding this. Not sure if we want to tackle it as part of this API rev.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 7 above)
Comment (nydickmsft) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1593]: It's a string, that must be parsable as an integer.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 1 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:651]: Fixed in next commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 2 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:712]: Fixed in next commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 3 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Updated to 200.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 4 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Fixed in next commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 5 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Fixed in next commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 6 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Fixed in next commit.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 7 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json:1593]: string in the RP based on Dan's other comment.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 8 above)
Comment (brpanask) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: Updated.

File: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json (hunk 3 above)
Comment (nydickmsft) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: When would we ever return 200?   If there's no body (which I think is the case), we'd return either 202 or 204, right?

Comment (omzevall) [reply specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2020-10-01/storagecache.json]: I think 200 is a required (by the validator?) return code for async ops. I know there are a few other 200 OKs listed in other endpoints that are unlikely to ever return it (204 No Content instead).

**Pull request merged**